backend/packages/Upgrade/src/api/services/QueryService.ts

```diff
@@ -66,6 +66,12 @@ export class QueryService {
 
     const promiseResult = await Promise.all(promiseArray);
     const experiments: Experiment[] = [];
+
+    // checks for archieve state experiment
+    if (promiseResult[0].experiment?.state === EXPERIMENT_STATE.ARCHIVED) {
+      return await this.getArchivedStats(queryIds, logger);
```

Comment:
You don't need to `await` here. Just return the promise

Reply:
@RidhamShah Can you please resolve this.